### PR TITLE
Prevent duplicate issues on GitHub reports

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

File: src/reporter/github.js [hunk 1]
```diff
@@ -25,173 +32,145 @@ export default class GitHub {
...
-      await this.openIssue(issue);
-    }
+        return logger.info(`🤖 Created GitHub issue #${createdIssue.number} "${title}": ${createdIssue.html_url}`);
+      }
 
-    const managedLabelsNames = this.MANAGED_LABELS.map(label => label.name);
-    const [managedLabel] = issue.labels.filter(label => managedLabelsNames.includes(label.name)); // it is assumed that only one specific reason for failure is possible at a time, making managed labels mutually exclusive
+      if (issue.state == GitHub.ISSUE_STATE_CLOSED) {
+        await this.openIssue(issue);
+        logger.info(`🤖 Reopen issue #${issue.number}: ${issue.html_url}`);
```
Comment (MattiSG) [inline src/reporter/github.js:157]: Do we really need to log this?
```suggestion
        logger.info(`Reopened issue #${issue.number}: ${issue.html_url}`);
```

File: src/reporter/github.js [hunk 2]
```diff
@@ -25,173 +32,145 @@ export default class GitHub {
...
-    const labelsNotManagedToKeep = issue.labels.map(label => label.name).filter(label => !managedLabelsNames.includes(label));
+      const labelsNotManagedToKeep = issue.labels.map(label => label.name).filter(label => !managedLabelsNames.includes(label));
 
-    await this.setIssueLabels({ issue, labels: [ label, ...labelsNotManagedToKeep ] });
-    await this.addCommentToIssue({ issue, comment: description });
+      await this.setIssueLabels({ issue, labels: [ label, ...labelsNotManagedToKeep ] });
+      await this.addCommentToIssue({ issue, comment: description });
+      logger.info(`🤖 Updated issue #${issue.number}: ${issue.html_url}`);
+    } catch (error) {
+      logger.error(`🤖 Failed to handle issue "${title}": ${error.message}`);
```
Comment (MattiSG) [inline src/reporter/github.js:173]:
```suggestion
      logger.error(`Failed to update issue "${title}": ${error.message}`);
```

**Review by MattiSG — APPROVED** [review]

File: src/reporter/github.js [hunk 3]
```diff
@@ -25,173 +32,145 @@ export default class GitHub {
...
+        for (const label of missingLabels) {
+          await this.createLabel({ /* eslint-disable-line no-await-in-loop */
+            name: label.name,
+            color: label.color,
+            description: `${label.description} ${MANAGED_BY_OTA_MARKER}`,
+          });
+        }
       }
+    } catch (error) {
+      logger.error(`🤖 Failed to handle repository labels: ${error.message}`);
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply src/reporter/github.js:52]: It is just legacy

File: src/reporter/github.js [hunk 4]
```diff
@@ -25,173 +32,145 @@ export default class GitHub {
...
+      if (!openedIssue) {
+        return;
+      }
 
-    await this.addCommentToIssue({ issue: openedIssue, comment });
+      await this.addCommentToIssue({ issue: openedIssue, comment });
+      await this.closeIssue(openedIssue);
 
-    return this.closeIssue(openedIssue);
+      return logger.info(`🤖 Closed issue #${openedIssue.number}: ${openedIssue.html_url}`);
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply src/reporter/github.js:139]: I find it useful to log the changes made to the reported issues.

**Pull request merged**